### PR TITLE
trace: fix pointer cross the border

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
VELAPLATFO-2058

Change-Id: I4cfa400c90d9421dfb565de3536b8378a28ae3fb
Signed-off-by: zhuyanlin <zhuyanlin1@xiaomi.com>

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: system/trace/trace.c [hunk 1]
```diff
@@ -735,27 +735,30 @@ static int trace_cmd_print(int index, int argc, FAR char **argv,
 
   /* Parse the setting parameters */
 
-  if (argv[index][0] == '-' || argv[index][0] == '+')
+  while (argv[index])
```
Comment (xiaoxiang781216) [inline system/trace/trace.c:738]: don't need while loop, let's check index < argc instead.

**Review by xiaoxiang781216 — CHANGES_REQUESTED** [review]

File: system/trace/trace.c (hunk 1 above)
Comment (xiaoxiang781216) [reply system/trace/trace.c:738]: and let's fix trace_cmd_switch too

File: system/trace/trace.c [hunk 2]
```diff
@@ -485,29 +485,32 @@ static int trace_cmd_switch(int index, int argc, FAR char **argv,
 
   /* Parse the setting parameters */
 
-  if (argv[index][0] == '-' || argv[index][0] == '+')
+  while (index < argc)
```
Comment (pkarashchenko) [inline system/trace/trace.c:488]:
```suggestion
  if (index < argc)
```

File: system/trace/trace.c [hunk 3]
```diff
@@ -735,29 +738,32 @@ static int trace_cmd_print(int index, int argc, FAR char **argv,
...
```
Comment (pkarashchenko) [inline system/trace/trace.c:741]:
```suggestion
  if (index < argc)
```

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

File: system/trace/trace.c [hunk 4]
```diff
@@ -538,7 +541,7 @@ static int trace_cmd_syscall(int index, int argc, FAR char **argv,
 
   /* Parse the setting parameters */
 
-  while (index < argc)
+ while (index < argc)
```
Comment (pkarashchenko) [inline system/trace/trace.c:544]:
```suggestion
  while (index < argc)
```

**Review by pkarashchenko — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**